jmx_prometheus_javaagent/src/main/java/io/prometheus/jmx/JavaAgent.java

```diff
@@ -16,105 +16,78 @@
-                    + // port
-                    "(.+)"; // config file
-
-    private static final String DEFAULT_HOST = "0.0.0.0";
-
-    private static HTTPServer httpServer;
-
-    public static void agentmain(String agentArgument, Instrumentation instrumentation)
-            throws Exception {
+    /**
+     * Java agent main
+     *
+     * @param agentArgument agentArgument
+     * @param instrumentation instrumentation
+     */
+    public static void agentmain(String agentArgument, Instrumentation instrumentation) {
         premain(agentArgument, instrumentation);
     }
 
-    public static void premain(String agentArgument, Instrumentation instrumentation)
-            throws Exception {
+    /**
+     * Java agent premain
+     *
+     * @param agentArgument agentArgument
+     * @param instrumentation instrumentation
+     */
+    public static void premain(String agentArgument, Instrumentation instrumentation) {
         try {
-            Config config = parseConfig(agentArgument);
+            Arguments arguments = Arguments.parse(agentArgument);
+            File file = new File(arguments.getFilename());
 
             new BuildInfoMetrics().register(PrometheusRegistry.defaultRegistry);
             JvmMetrics.builder().register(PrometheusRegistry.defaultRegistry);
-            new JmxCollector(new File(config.file), JmxCollector.Mode.AGENT)
+            new JmxCollector(new File(arguments.getFilename()), JmxCollector.Mode.AGENT)
                     .register(PrometheusRegistry.defaultRegistry);
 
-            String host = config.host != null ? config.host : DEFAULT_HOST;
-
-            httpServer =
-                    new HTTPServerFactory()
-                            .createHTTPServer(
-                                    InetAddress.getByName(host),
-                                    config.port,
-                                    PrometheusRegistry.defaultRegistry,
-                                    new File(config.file));
+            switch (arguments.getMode()) {
+                case HTTP:
+                    {
+                        HTTPServerFactory.getInstance()
+                                .createHTTPServer(
+                                        InetAddress.getByName(arguments.getHost()),
+                                        arguments.getPort(),
+                                        PrometheusRegistry.defaultRegistry,
+                                        file);
+
+                        break;
+                    }
+                case OPEN_TELEMETRY:
+                    {
+                        OpenTelemetryExporterFactory.getInstance()
+                                .createOpenTelemetryExporter(
+                                        PrometheusRegistry.defaultRegistry, file);
+
+                        break;
+                    }
+                default:
+                    {
+                        throw new RuntimeException("Undefined mode");
```

Comment:
This error message isn't very intuitive. What do you think of printing something like this?
```
"Failed to start the Prometheus JMX exporter: Either specify a port for the Prometheus scrape endpoint, or enable OpenTelemetry mode."
```

Also, maybe we should not print the stack trace here. We could just print the error message to stderr and then exit:

```
System.err.println("...");
System.exit(-1);
```